mapiproxy/servers/default/asyncemsmdb/dcesrv_asyncemsmdb.c

```diff
@@ -0,0 +1,1230 @@
+		}
+		retvals[i] = retval;
+	}
+
+	return MAPI_E_SUCCESS;
+}
+
+
+
+/**
+   \details Process a TableModified event on a ContentsTable for row modified specific event type
+
+   \param mem_ctx pointer to the memory context
+   \param p pointer to the private asyncemsmdb data
+   \param s pointer to the array of subscriptions for this session
+   \param folderId the folder in which the tablemodified event occurred
+
+   \note This first iteration has the following limitations:
+   - Only TABLE_ROW_MODIFIED case are handled
+   - Only handle notifications within Inbox folder (openchangedb folders)
+
+   \return 0 on success, otherwise -1
+ */
+static int process_tablemodified_contentstable_notification(TALLOC_CTX *mem_ctx,
+							    struct asyncemsmdb_private_data *p,
+							    struct mapistore_notification_subscription *s,
+							    uint64_t folderId)
+{
+	int				i;
+	struct EcDoRpc_MAPI_REPL	reply;
+	DATA_BLOB			payload;
+	struct SPropTagArray		SPropTagArray;
+	void				**data_pointers;
+	enum MAPISTATUS			*retvals;
+	enum MAPISTATUS			retval;
+	int				ret;
+	uint32_t			prop_idx;
+	int				flagged = 0;
+	uint32_t			property;
+	void				*data = NULL;
+	enum ndr_err_code		ndr_err_code;
+	struct ndr_push			*ndr;
+	char				*mapistoreURI = NULL;
+
+	/* Looking */
+	for (i = 0; i < s->v.v1.count; i++) {
+		if (s->v.v1.subscription[i].flags & sub_TableModified) {
+			if ((s->v.v1.subscription[i].flags & sub_WholeStore) ||
+			    (s->v.v1.subscription[i].fid == folderId)) {
+				memset(&reply, 0, sizeof(struct EcDoRpc_MAPI_REPL));
+				reply.opnum = op_MAPI_Notify;
+				reply.error_code = MAPI_E_SUCCESS;
+				reply.u.mapi_Notify.NotificationHandle = s->v.v1.subscription[i].handle;
+				reply.u.mapi_Notify.LogonId = 0;
+				reply.u.mapi_Notify.NotificationType = fnevMbit | fnevSbit | fnevTableModified;
+				reply.u.mapi_Notify.NotificationData.ContentsTableChange.TableEvent = 0x5; /* TABLE_ROW_MODIFED */
+				reply.u.mapi_Notify.NotificationData.ContentsTableChange.ContentsTableChangeUnion.ContentsRowModifiedNotification.FID = folderId;
+				reply.u.mapi_Notify.NotificationData.ContentsTableChange.ContentsTableChangeUnion.ContentsRowModifiedNotification.MID = 0x0;
+				reply.u.mapi_Notify.NotificationData.ContentsTableChange.ContentsTableChangeUnion.ContentsRowModifiedNotification.Instance = 0x0;
+				reply.u.mapi_Notify.NotificationData.ContentsTableChange.ContentsTableChangeUnion.ContentsRowModifiedNotification.InsertAfterFID = 0x0;
+				reply.u.mapi_Notify.NotificationData.ContentsTableChange.ContentsTableChangeUnion.ContentsRowModifiedNotification.InsertAfterMID = 0x0;
+				reply.u.mapi_Notify.NotificationData.ContentsTableChange.ContentsTableChangeUnion.ContentsRowModifiedNotification.InsertAfterInstance = 0x0;
+
+				SPropTagArray.cValues = s->v.v1.subscription[i].count;
+				SPropTagArray.aulPropTag = (enum MAPITAGS *)s->v.v1.subscription[i].properties;
+
+				data_pointers = talloc_array(mem_ctx, void *, SPropTagArray.cValues);
```

Comment:
There is already a method called `talloc_zero_array` that I would use instead of talloc_array+memset (it's the same as you are doing here, but more clear one call that two, isn't it?)